samples/unit-tests/series/seriestypes/demo.js

```diff
@@ -39,4 +39,42 @@ Object.keys(Highcharts.seriesTypes).forEach(function (type) {
             );
         });
     }
+});
+
+QUnit.test('#13277: Event listener memory leak', assert => {
+    Object.keys(Highcharts.seriesTypes).forEach(type => {
+        if (
+            !('linkedTo' in Highcharts.defaultOptions.plotOptions[type]) &&
+            type !== 'scatter3d' &&
+            type !== 'map' && type !== 'mapline' // Transform error on redraw
+        ) {
+            const chart = Highcharts.chart('container', {
+                chart: {
+                    type: type
+                },
+                series: [{
+                    name: 'Test series'
+                }]
+            });
+
+            const eventCount = el => {
+                let count = 0;
+                //eslint-disable-next-line
+                for (const t in el.hcEvents) {
+                    count += el.hcEvents[t].length;
+                }
+                return count;
+            };
+
+            const before = eventCount(chart.series[0]);
+            const beforeChart = eventCount(chart);
+
+            chart.series[0].update({
+                data: []
+            });
+
+            assert.ok(before === eventCount(chart.series[0]), `No leakage into series.hcEvents for ${type} series`);
```

Comment:
Best practices:
* Use `assert.strictEqual` instead of `assert.ok` and `===`.
* Include the word "should" in the description. This way it is clear what we _expect_ to see, rather than what we actually see, which is wrong when the test fails.